api/v1alpha2/vpcendpoint_types.go

```diff
@@ -118,6 +118,12 @@ type VpcEndpointSpec struct {
 
 	// +kubebuilder:validation:Optional
 
+	// AWSCredentialOverride is a Kubernetes secret containing AWS credentials for the operator to use for reconciling
+	// this specific vpcendpoint Custom Resource
+	AWSCredentialOverrideRef *corev1.SecretReference `json:"awsCredentialOverrideRef,omitempty"`
```

Comment:
If we are restricting to a single namespace, should this just be a string then?

Reply:
I was thinking that the operator code would be unlocked, but for OSD purposes we could lock down the K8s RBAC so that we only allow reading `openshift-aws-vpce-operator` secrets until we need anything more